### ISSUE TITLE
Give users the opportunity to self-delete accounts

### ISSUE THREAD

**Issue by andreasgeissner** [DSpace/DSpace#8711]
**Is your feature request related to a problem? Please describe.**
According to European privacy laws, users have the right to have their personal data deleted from information systems. Currently, users have to contact the repository administration to initiate removal of accounts and the personal data.

**Describe the solution you'd like**
A button within the user's profile to allow users to self-delete their account and the attached personal data. This button should work for users that have not submitted their own datasets. 

**Describe alternatives or workarounds you've considered**
As stated above, it is possible for users to request account deletion at the

**Comment by tdonohue** [comment]
@arvoConsultores has asked to claim this ticket. Assigned to their team.

**Comment by aroman-arvo** [comment]
What about users that has submitted an item? maybe their finished submissions should be assigned to collection or general administrator?

**Comment by andreasgeissner** [comment]
In my original post, I suggested that the button should only work for users that have not submitted any items. I don't know if that should be up for configuration ´.

**Comment by aroman-arvo** [comment]
I know it was proposed only to unsubmitted users, but before I start makin changes I would like to launch an open question. @tdonohue European privacy laws should be aplied to users with submitted items? if yes how? reassiging it to collection or general administrator? Keeping user but overwriting his personal data with "deleted-user-18"?

**Comment by tdonohue** [comment]
@aroman-arvo : My understanding is that in DSpace 7 it is now possible to delete EPersons _even if they have submitted content_.  See the fixes in #2229 made by @pnbecker .  

I cannot answer the question about European privacy laws as I'm not well versed in that.  But, if it does require that people be able to *self-delete*, then it seems reasonable to allow a logged in person to self-delete their own account.  Since it's already possible to delete EPersons who are referenced elsewhere, then it might be as simple as allowing any EPerson to delete their own account.

It's possible @pnbecker has a better answer here as he's worked in this area of the code much in the past & he is better versed in GDPR and similar than I am.

**Comment by pnbecker** [comment]
I think we must be able to delete user. I don't think that users must be able to do this themselves. But I'm not a lawyer and I might be wrong.

**Comment by andreasgeissner** [comment]
I agree with @pnbecker, but I'm also no lawyer. I would be careful with allowing users that have submitted anything to delete themselves. They have, after all, entered into a contract with the repository provider and the provider may want to be able to decide whether they still need information on who this contract is with (according the the GDPR, providers are only required to delete personal information if there is no reason to keep it)., and delete themsleves if the decide they don't. 

That's also why I had limited the issue to users that have not submitted datasets.

**Comment by aroman-arvo** [comment]
It is clear then. By default only users without submitted items can auto-delete their user. It is the responsibility of the institution to manage users who have submitted items. 
With the work made in #2803 it should be easy.

**Comment by tdonohue** [comment]
I want to clarify this ticket a bit further.  I understood @pnbecker 's [prior comment](https://github.com/DSpace/DSpace/issues/8711#issuecomment-1749297742) as saying "The current behavior may be good enough" **because administrators already have the ability to delete users.**   Are we certain that Users *must* be able to delete themselves? Or is it simply enough to document that they can contact their Administrator if they want their account to be permanently deleted?  DSpace already allows users to be deleted, but they simply need to ask their Administrator to make that happen.

I'm also not a lawyer (obviously), but from what I see on GDPR from other sites, the "Right to Erasure" (or right to be forgotten) simply gives users the "right to have your data erased, without undue delay **by the data controller**".  Quote taken from https://www.dataprotection.ie/en/individuals/know-your-rights/right-erasure-articles-17-19-gdpr

Here's the exact text from GDPR: https://gdpr-info.eu/art-17-gdpr/
> The data subject shall have the right to obtain from the controller the erasure of personal data concerning him or her without undue delay and the controller shall have the obligation to erase personal data without undue delay where one of the following grounds applies ...

That implies to me that DSpace doesn't need to allow users to delete themselves.  We just need to allow Administrators to be able to delete users, if a user requests it (and we already have that feature in 7.x)

So, I'd like to ask whether we feel the current behavior is "good enough" to align with GDPR.  I'm not a lawyer, but I've yet to find exact text in GDPR which **requires us to allow users to delete themselves**.

**Comment by pnbecker** [comment]
I think the current behavior is good enough.

**Comment by aroman-arvo** [comment]
I think It is sufficient to comply with GDPR. 
If there is no interest, for usability reasons, the issue could be closed.

**Comment by andreasgeissner** [comment]
The issue was mainly for usability so that not all users have to contact central support and central support does not have to handle such requests. I agree, to comply with GDPR, the current behavior is good enough.
